identity-service/src/notifications/sendNotificationEmails.js

```diff
@@ -63,49 +63,60 @@ async function processEmailNotifications (expressApp, audiusLibs) {
     let liveUsersWithPendingAnnouncements = []
     let dailyUsersWithPendingAnnouncements = []
     let weeklyUsersWithPendingAnnouncements = []
+
+    const timeBeforeAnnouncementsLoop = Date.now()
+    logger.debug(`processEmailNotifications | time before looping over announcements | ${timeBeforeAnnouncementsLoop} | ${appAnnouncements.length} announcements`)
     for (var announcement of appAnnouncements) {
       let announcementDate = moment(announcement['datePublished'])
       let timeSinceAnnouncement = moment.duration(currentTime.diff(announcementDate)).asHours()
       let announcementEntityId = announcement['entityId']
       let id = announcement['id']
-      let usersCreatedAfterAnnouncement = await models.User.findAll({
+      let usersCreatedBeforeAnnouncement = await models.User.findAll({
         attributes: ['blockchainUserId'],
         where: {
           createdAt: { [models.Sequelize.Op.lt]: moment(announcementDate) }
         }
       }).map(x => x.blockchainUserId)
 
-      for (var user of usersCreatedAfterAnnouncement) {
-        let userNotificationQuery = await models.Notification.findOne({
-          where: {
-            isViewed: true,
-            userId: user,
-            type: notificationTypes.Announcement,
-            entityId: announcementEntityId
-          }
-        })
-        if (userNotificationQuery) {
-          continue
+      const userIdsToExcludeForAnnouncement = await models.Notification.findAll({
+        attributes: ['userId'],
+        where: {
+          isViewed: true,
+          userId: { [models.Sequelize.Op.in]: usersCreatedBeforeAnnouncement },
+          type: notificationTypes.Announcement,
+          entityId: announcementEntityId
         }
+      })
+      const userIdSetToExcludeForAnnouncement = new Set(userIdsToExcludeForAnnouncement)
```

Comment:
Think you might need to map userIdsToExcludeForAnnouncement to userId
ie. `userIdsToExcludeForAnnouncement.map(u => u.userId)`